mobile/envoy_build_config/extension_registry.cc

```diff
@@ -38,70 +52,113 @@
-  Envoy::Extensions::Http::HeaderFormatters::PreserveCase::
+  Common::Http::MatchDelegate::Factory::forceRegisterSkipActionFactory();
+  Common::Http::MatchDelegate::forceRegisterMatchDelegateConfig();
+  ExtensionRegistryPlatformAdditions::registerFactories();
+  Extensions::Clusters::DynamicForwardProxy::forceRegisterClusterFactory();
+  Extensions::Compression::Brotli::Decompressor::forceRegisterBrotliDecompressorLibraryFactory();
+  Extensions::Compression::Gzip::Decompressor::forceRegisterGzipDecompressorLibraryFactory();
+  Extensions::Http::HeaderFormatters::PreserveCase::
       forceRegisterPreserveCaseFormatterFactoryConfig();
-  Envoy::Extensions::HttpFilters::AlternateProtocolsCache::
+  Extensions::Http::HeaderValidators::EnvoyDefault::forceRegisterHeaderValidatorFactoryConfig();
+  Extensions::Http::OriginalIPDetection::Xff::forceRegisterXffIPDetectionFactory();
+  Extensions::HttpFilters::AlternateProtocolsCache::
       forceRegisterAlternateProtocolsCacheFilterFactory();
-  Envoy::Extensions::HttpFilters::Assertion::forceRegisterAssertionFilterFactory();
-  Envoy::Extensions::HttpFilters::Decompressor::forceRegisterDecompressorFilterFactory();
-  Envoy::Extensions::HttpFilters::BufferFilter::forceRegisterBufferFilterFactory();
-  Envoy::Extensions::HttpFilters::DynamicForwardProxy::
-      forceRegisterDynamicForwardProxyFilterFactory();
-  Envoy::Extensions::HttpFilters::LocalError::forceRegisterLocalErrorFilterFactory();
-  Envoy::Extensions::HttpFilters::PlatformBridge::forceRegisterPlatformBridgeFilterFactory();
-  Envoy::Extensions::HttpFilters::RouteCacheReset::forceRegisterRouteCacheResetFilterFactory();
-  Envoy::Extensions::HttpFilters::RouterFilter::forceRegisterRouterFilterConfig();
-  Envoy::Extensions::HttpFilters::NetworkConfiguration::
-      forceRegisterNetworkConfigurationFilterFactory();
-  Envoy::Extensions::KeyValue::forceRegisterPlatformKeyValueStoreFactory();
-  Envoy::Extensions::NetworkFilters::HttpConnectionManager::
+  Extensions::HttpFilters::Assertion::forceRegisterAssertionFilterFactory();
+  Extensions::HttpFilters::BufferFilter::forceRegisterBufferFilterFactory();
+  Extensions::HttpFilters::Decompressor::forceRegisterDecompressorFilterFactory();
+  Extensions::HttpFilters::DynamicForwardProxy::forceRegisterDynamicForwardProxyFilterFactory();
+  Extensions::HttpFilters::LocalError::forceRegisterLocalErrorFilterFactory();
+  Extensions::HttpFilters::NetworkConfiguration::forceRegisterNetworkConfigurationFilterFactory();
+  Extensions::HttpFilters::PlatformBridge::forceRegisterPlatformBridgeFilterFactory();
+  Extensions::HttpFilters::RouteCacheReset::forceRegisterRouteCacheResetFilterFactory();
+  Extensions::HttpFilters::RouterFilter::forceRegisterRouterFilterConfig();
+  Extensions::HttpFilters::SocketTag::forceRegisterSocketTagFilterFactory();
+  Extensions::HttpFilters::TestAccessor::forceRegisterTestAccessorFilterFactory();
```

Comment:
Yes, it's only used for Swift tests right now though: https://github.com/envoyproxy/envoy/blob/main/mobile/test/swift/integration/test_extensions.cc

I'm not sure if it's problematic to register these extensions for non-tests, do you?

Reply:
Ah yes, this does increase binary size by ~1.9% according to the `mobile_perf` CI jobs. Let me see if I can only register these for tests in this PR to avoid any binary size increase.